### PR TITLE
Allow large UDT buffers for netfx

### LINKED ISSUE #329: UDT parameters are limited to 64K
This is an issue because serialized spatial types can easily be bigger than this. See https://github.com/aspnet/EntityFrameworkCore/issues/18813.

@Wraith2 pointed to the place in the code where this limitation exists: https://github.com/dotnet/SqlClient/blob/d0672d2c4dfd0f09ba49f09d605b827366c61b1b/src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs#L9027-L9029

/cc @bricelam

### PR DESCRIPTION
**Pull request by alsicbv**
fixes  #329 for netfx

The fix in #340 was only for netcore. Issue #329 still exists when using EF core on Net Framework.

/cc @ajcvickers @aprognimak  @Wraith2

### CODE REVIEW

**Comment by Wraith2** [comment]
It'll do the job. I'd prefer the new exception type and message from the other PR to come over as well but it should almost never be thrown so it isn't a huge probem.

**Comment by alsicbv** [comment]
> 
> 
> It'll do the job. I'd prefer the new exception type and message from the other PR to come over as well but it should almost never be thrown so it isn't a huge probem.

@Wraith2 What should be done to have this reviewed and merged? It seems I don't have rights to request a review.

**Comment by Wraith2** [comment]
You just wait i'm afraid.

**Comment by cheenamalhotra** [comment]
Hi @alsicbv 

As @Wraith2 mentioned before, please include all necessary changes from NetCore PR here. We are only waiting for that in order to get this merged.

**Comment by Wraith2** [comment]
In all fairness to @alsicbv as a contributor I can only make suggestions and it wasn't clear that it was my suggestion that was being waited for. I think both he and I assumed that the PR was waiting for review from the owner side.

**Comment by cheenamalhotra** [comment]
Oops!
You do bring valuable suggestions @Wraith2 , I'll make a note to highlight the same next time!

**Comment by alsicbv** [comment]
> 
> 
> Oops!
> You do bring valuable suggestions @Wraith2 , I'll make a note to highlight the same next time!

Ok, it wasn't clear to me... Thanks for clarifying, I'm not working tomorrow so I'll do it on Friday!

**Comment by alsicbv** [comment]
> 
> 
> Hi @alsicbv
> 
> As @Wraith2 mentioned before, please include all necessary changes from NetCore PR here. We are only waiting for that in order to get this merged.

I've made all changes from the original PR

File: src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.resx [hunk 1]
```diff
@@ -4527,4 +4527,7 @@
   <data name="TCE_AttestationProtocolNotSpecifiedForGeneratingEnclavePackage" xml:space="preserve">
     <value>Error occurred when generating enclave package. Attestation Protocol has not been specified in the connection string, but the query requires enclave computations.</value>
   </data>
-</root>
+  <data name="SQLUDT_InvalidSize" xml:space="preserve">
+    <value>UDT size must be less than {1}, size: {0}</value>
+  </data>
+</root>
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.resx:4533]: Nit: EOF line

Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/netfx/src/Resources/Strings.resx:4533]: Nvm, I think it's autogenerated.

**Pull request merged**